multicluster/test/e2e/service_test.go

```diff
@@ -47,32 +47,39 @@ func testServiceExport(t *testing.T, data *TestData) {
 // we create a nginx in on cluster(east), and try to curl it in another cluster(west).
 // If we got status code 200, it means that the resources is exported by the east cluster
 // and imported by the west cluster.
-// TODO(yang): reorg test function contents
-func (data *TestData) testServiceExport(t *testing.T) {
-	podName := randName("test-nginx-")
-	clientPodName := "test-service-client"
-
-	if err := createPodWrapper(t, data, westCluster, multiClusterTestNamespace, podName, nginxImage, "nginx", nil, nil, nil, nil, false, nil); err != nil {
-		t.Fatalf("Error when creating nginx Pod in west cluster: %v", err)
-	}
-	defer deletePodWrapper(t, data, westCluster, multiClusterTestNamespace, podName)
-
-	if err := createPodWrapper(t, data, eastCluster, multiClusterTestNamespace, podName, nginxImage, "nginx", nil, nil, nil, nil, false, nil); err != nil {
-		t.Fatalf("Error when creating nginx Pod in east cluster: %v", err)
-	}
-	defer deletePodWrapper(t, data, eastCluster, multiClusterTestNamespace, podName)
 
+func (data *TestData) setupTestServices(t *testing.T) {
 	if _, err := data.createService(westCluster, westClusterTestService, multiClusterTestNamespace, 80, 80, corev1.ProtocolTCP, map[string]string{"app": "nginx"}, false,
 		false, corev1.ServiceTypeClusterIP, nil, nil); err != nil {
 		t.Fatalf("Error when creating Servie %s in west cluster: %v", westClusterTestService, err)
 	}
-	defer deleteServiceWrapper(t, testData, westCluster, multiClusterTestNamespace, westClusterTestService)
 
 	if _, err := data.createService(eastCluster, eastClusterTestService, multiClusterTestNamespace, 80, 80, corev1.ProtocolTCP, map[string]string{"app": "nginx"}, false,
 		false, corev1.ServiceTypeClusterIP, nil, nil); err != nil {
 		t.Fatalf("Error when creating Servie %s in east cluster: %v", eastClusterTestService, err)
 	}
-	defer deleteServiceWrapper(t, testData, eastCluster, multiClusterTestNamespace, eastClusterTestService)
+}
+
+func (data *TestData) tearDownTestServices(t *testing.T) {
+	deleteServiceWrapper(t, testData, westCluster, multiClusterTestNamespace, westClusterTestService)
+	deleteServiceWrapper(t, testData, eastCluster, multiClusterTestNamespace, eastClusterTestService)
+}
+
+func (data *TestData) getAndProberFromCluster(t *testing.T, clusterName string){
```

Comment:
I'm confused by this function.... it takes a clusterName and have `westClusterTestService` inside hardcoded?

Reply:
Let me fix it. Thanks!